ee/presencedetection/presencedetection.go

```diff
@@ -32,8 +33,13 @@ func (d *detector) Detect(reason string) (bool, error) {
 // DetectPresence checks if the user is present by detecting the presence of a user.
 // It returns the duration since the last detection.
 func (pd *PresenceDetector) DetectPresence(reason string, detectionInterval time.Duration) (time.Duration, error) {
-	pd.mutext.Lock()
-	defer pd.mutext.Unlock()
+	// using try lock here because we don't want presence detections to queue up,
+	// in the event that the users presses cancel, if the request were queued up, it would
+	// request the presence detection again
+	if !pd.mutex.TryLock() {
```

Comment:
actually, this seems racy, may need 2 mutexes here, one for the lastDetection value and one for actually prompting the user